src/Emulator/Peripherals/Peripherals/Timers/OpenTitan_Timer.cs

```diff
@@ -0,0 +1,344 @@
+        }
+
+        public void Reset()
+        {
+            registers.Reset();
+            for(int i = 0; i < numberOfHarts; i++)
+            {
+                for(int j = 0; j < numberOfTimers; j++) 
+                {
+                    Timers[i, j].Reset();
+                    GetIrqForHart(i, j).Set(false);
+                }
+            }
+        }
+
+        public void WriteDoubleWord(long offset, uint value)
+        {
+            registers.Write(offset, value);
+        }
+
+        public String PrintInnerTimerStatus(int hartIdx, int timerIdx)
+        {
+            ComparingTimer timer = Timers[hartIdx, timerIdx];
+            return String.Format("ComparingTimer(Enabled:{0},EventEnables:{1},Compare:0x{2:X},Value:0x{3:X},Divider:0x{4:X})",
+                timer.Enabled, timer.EventEnabled, timer.Compare, timer.Value, timer.Divider);
+        }
+
+        public long Size => 0x2000; // Change size to allow for additional harts (0x2000 allows for 31 harts)
+        public IReadOnlyDictionary<int, IGPIO> Connections { get; }
+
+        private IGPIO GetIrqForHart(int hartIdx, int timerIdx)
+        {
+            return IRQs[hartIdx * numberOfTimers + timerIdx];
+        }
+
+        private void AddTimerControlRegisters(Dictionary<long, DoubleWordRegister> registersMap, long address, int hartCount)
+        {   
+            var maximumControlRegCount = (int)Math.Ceiling((hartCount) / 32.0);
+            for(int timerCtrlIdx = 0; timerCtrlIdx < maximumControlRegCount; timerCtrlIdx++)
+            {
+                var offset = timerCtrlIdx * 4;
+                int hartBitCount = (hartCount < 32) ? hartCount : 32;
+                hartCount -= hartBitCount;
+                var controlRegister = new DoubleWordRegister(this);
+                for(int hartBitIdx = 0; hartBitIdx < hartBitCount; hartBitIdx++)
+                {
+                    int timerEnablesIdx = timerCtrlIdx * 32 + hartBitIdx;
+                    var flagName = String.Format("CONTROL{0}_{1}", timerCtrlIdx, timerEnablesIdx);
+                    this.Log(LogLevel.Noisy, "Add Enable bit for {0} as field {1}", timerEnablesIdx, flagName);
+                    timerEnables[timerEnablesIdx] = controlRegister.DefineFlagField(hartBitIdx, name: flagName, writeCallback: (idx, val) => 
+                    {
+                        this.Log(LogLevel.Noisy, "Set Enable bit for {0} to {1}", timerEnablesIdx, val);
+                        for(int i = 0; i < numberOfTimers; i++) 
+                        {
+                            Timers[timerEnablesIdx, i].Enabled = val;
+                        }
+                    });
+                }
+                this.Log(LogLevel.Noisy, "Add REG: CTRL[0x{0:X}]", address + offset);
+                registersMap.Add(address + offset, controlRegister);
+            }
+        }
+
+        private void AddTimerRegisters(Dictionary<long, DoubleWordRegister> registersMap, long address, int hartId)
```

Comment:
Is there a point of having the `address` register? It seems to be `0` anyway.

Reply:
I will remove it, unused offset since it is always assumed to be from 0